packages/core/src/application_init.ts

```diff
@@ -59,8 +59,8 @@ import {noop} from './util/noop';
  * through DI.
  *
  * ```
- *  function initializeApp(httpClient: HttpClient): Observable<any> {
- *   return httpClient.get("https://someUrl.com/api/user")
+ *  function initializeApp(httpClient: HttpClient): () => Observable<any> {
+ *   return (): Observable<any> => httpClient.get("https://someUrl.com/api/user")
```

Comment:
I believe the return type is not necessary here:

```suggestion
 *   return () => httpClient.get("https://someUrl.com/api/user")
```